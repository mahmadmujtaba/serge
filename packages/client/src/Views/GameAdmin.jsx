import React, { Component } from "react";
import { CHAT_CHANNEL_ID, expiredStorage, LOCAL_STORAGE_TIMEOUT } from "../consts";
import MessageCreatorChatChannel from "../Components/MessageCreatorChatChannel";
import MessagesListChatChannel from "./MessagesListChatChannel";
import MessagesListRenderProp from "./MessagesListRenderProp";
import { PlayerStateContext } from "../Store/PlayerUi";
import "../scss/App.scss";

class GameAdmin extends Component {
  static contextType = PlayerStateContext;

  constructor(props, context) {
    super(props);
    const [ state ] = context;
    this.state = {
      activeTab: Object.keys(state.channels)[0],
      allMarkedRead: false,
      showObjectives: false,
    };
  }

  componentDidMount() {
    const [ state ] = this.context;
    let channelLength = Object.keys(state.chatChannel.messages).length;

    if (channelLength) {
      this.setState({allMarkedRead: false});
    }
  }

  markAllAsRead = () => {
    const [ state ] = this.context;
    state.chatChannel.messages.forEach((message) => {
      expiredStorage.setItem(state.currentWargame + message._id, "read", LOCAL_STORAGE_TIMEOUT);
    });
    this.setState({
      allMarkedRead: true,
    })
  };

  render() {
    const [ state ] = this.context;
    return (
      <div className="contain-game-admin">
        <MessagesListRenderProp
          curChannel={CHAT_CHANNEL_ID}
          messages={state.chatChannel.messages}
          userId={`${state.wargameTitle}-${state.selectedForce}-${state.selectedRole}`}
          allMarkedRead={this.state.allMarkedRead}
          render={messages => (
            <MessagesListChatChannel
              messages={messages}
              markAllAsRead={this.markAllAsRead}
            />
          )}
        />

        <div className="new-message-creator wrap" data-tour="seventh-step">
          <MessageCreatorChatChannel
              curChannel={CHAT_CHANNEL_ID}
              schema={state.chatChannel.template}
          />
        </div>
      </div>
    );
  }
}

export default GameAdmin;
