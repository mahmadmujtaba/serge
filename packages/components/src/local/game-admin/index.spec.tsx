/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import GameAdmin from './index'

it('GameAdmin renders correctly', () => {
  const tree = renderer
    .create(<GameAdmin />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});