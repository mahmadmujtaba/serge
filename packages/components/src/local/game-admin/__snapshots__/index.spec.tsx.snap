// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameAdmin renders correctly 1`] = `
<div
  className="blue"
>
  This is an example component, the name provided to it was 
  test
</div>
`;
